scripts/globals/job_utils/dancer.lua

```diff
@@ -493,6 +492,13 @@ xi.job_utils.dancer.useWaltzAbility = function(player, target, ability, action)
     amtCured = (target:getStat(xi.mod.VIT) + player:getStat(xi.mod.CHR)) * statMultiplier + waltzInfo[3]
     amtCured = math.floor(amtCured * (1.0 + (math.min(50, player:getMod(xi.mod.WALTZ_POTENCY)) / 100)))
     -- TODO: Account for Waltz Potency Received
+    if player:getStatusEffect(xi.effect.CONTRADANCE) then
+        amtCured = amtCured * 2
+        -- slight delay to allow the effect to apply to all targets of divine waltz
+        player:timer(100, function(playerArg)
+            playerArg:delStatusEffect(xi.effect.CONTRADANCE)
```

Comment:
Don't like this, but there's not much we can do about it. Maybe we need some kind of self-removing listener? I think it's okay to leave it in like this for now with a TODO to improve it

Reply:
https://github.com/LandSandBoat/server/blob/ece956579adf1d2f9aa7f97552e2637a070a132c/src/map/utils/battleutils.cpp#L5675

Need something like that, but for abilities instead of spells. There is no correct way to do this in lua because it should not be happening in lua.

Reply:
BTW a TODO was added for this. Unsure what the hold label was added for

Reply:
> Unsure what the hold label was added for

_point back at own previous comment_

was so nobody merged ahead of conferring on if this was acceptable temporarily, exactly how temporary that would be, and who if anybody would be coming back to do it right. Worth mentioning that temporary measures are technical debt, and past iterations of the project added more technical debt than the GDP of some real world nations.

@claywar  is our resident DNC expert I think. If He's ok with this I'll remove the hold I placed.

Reply:
I realize that this runs for every target, would it be more appropriate to set the contradance effect duration to zero instead of a timer (still will run for every target) so it falls off immediately after the targetfind loop?